core/scripts/cli/sponsor/redeem.js

```diff
@@ -27,7 +28,8 @@ const redeem = async (web3, artifacts, emp) => {
   });
 
   const tokensToRedeem = toBN(toWei(input["numTokens"]));
-  console.log("You'll receive", fromWei(collateralPerToken.mul(toBN(tokensToRedeem)).div(scalingFactor)), "ETH");
+  const expectedCollateral = collateralPerToken.mul(toBN(tokensToRedeem)).div(scalingFactor);
```

Comment:
Not related to this PR, but wanted to get your thoughts.

Would it help for us to add a return value to functions that return money to the user, like redeem? In that case, we might be able to avoid some math in js by simulating the txn before sending it? Or do you think most of the math we do in js is unrelated to those sorts of cases?

Reply:
Definitely would be helpful in the redeem case. Otherwise it might be hard to get the exact rounding behavior in JS.

There's still going to be some math done in JS that we probably won't be able to push to the contract, such as computing the collateralization ratio if a withdrawal goes through, etc. But in those cases, being a little off in the rounding is probably not a big deal.

Reply:
Cool cool, I'll take a look at the contract and open an issue.